src/bm_sim/P4Objects.cpp

```diff
@@ -422,6 +422,57 @@ P4Objects::add_primitive_to_action(const Json::Value &cfg_primitive,
     if (type == "hexstr") {
       const auto value_hexstr = cfg_parameter["value"].asString();
       action_fn->parameter_push_back_const(Data(value_hexstr));
+    } else if (type == "parameters_vector") {
```

Comment:
is there any chance that this code and the "regular" parameter code can be unified in some way?

Reply:
I think there is no chance that this code can be unified with regular one because when we have parameter vector we go one step further, and i think we must go through for loop separately for parameter vector types.

Reply:
I need to think about this one more time, maybe I can solve this